### PR TITLE
Add functional Test for Public Covid Record AB#11877 AB#11899 AB#11901

### PR DESCRIPTION
**Pull request by pagustinqt20**
# Add functional Test for Public Covid Record [AB#11877](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/11877) [AB#11899](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/11899) [AB#11901](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/11901)

## Description
This PR includes the following changes :
1. Add functional Test for Public Covid Record. 
2. Add functional test for the fix of duplicate navigation bars when route have capital letter (ex. covidTest or vaccineCard).
3. Public COVID-19 Test Form References Vaccine Date
4. Fix the bug on ProofOfVaccination Functional Test

If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

## Testing

-   [ ] Unit Tests Updated
-   [x] Functional Tests Updated
-   [ ] Not Required

### UI Changes

**Yes**

image

### Browsers Tested

-   [x] Chrome
-   [ ] Safari
-   [ ] Edge
-   [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant. Include any specialized deployment steps here.

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

-   Fulfills Work Item Requirements

    -   The changes meet/implement story's acceptance criteria. Fixes identified problems if bug.
    -   Changes not directly related to the task requirements need to be documented on the PR.

-   Compilation / Tests

    -   The changes do not break compilation and/or unit or functional tests; unless other item addresses them specifically.

-   Logic Problems / Functional Behaviour

    -   The changes work as intended and do not introduce problems.

-   Performance

    -   The changes do not introduce obvious performance issues.

-   Documented Standards

    -   The changes adhere to the team's documented [Coding Standards](https://github.com/bcgov/healthgateway/wiki/standards).

-   Readability / Maintainability
    -   The changes can be easily understood/read and allow for future enhancements without major refactoring. Readability is preferred over "clever code".
    -   Reasoning for changes needs to be clearly articulated. Disagreements should be arbitrated by a third developer.

### CODE REVIEW

File: Testing/functional/tests/cypress/integration/ui/vaccineCard/publicCovidTest.js [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+    cy.get("[data-testid=btnEnter]").should("be.enabled", "be.visible").click();
+}
+
+describe("Public User - Covid19 Test Result Page", () => {
+    beforeEach(() => {
+        cy.enableModules([laboratoryModule, publicLaboratoryResultModule]);
+        cy.visit(covidTestUrl);
+    });
+
+    it("Public Vaccination Card - Cancel", () => {
```
Comment (MikeLyttle) [inline Testing/functional/tests/cypress/integration/ui/vaccineCard/publicCovidTest.js:70]: This is not the vaccination card.

File: Testing/functional/tests/cypress/integration/ui/vaccineCard/publicCovidTest.js [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+const dobMonthSelector =
+    "[data-testid=dateOfBirthInput] [data-testid=formSelectMonth]";
+const dobDaySelector =
+    "[data-testid=dateOfBirthInput] [data-testid=formSelectDay]";
+const dovYearSelector =
+    "[data-testid=dateOfVaccineInput] [data-testid=formSelectYear]";
+const dovMonthSelector =
+    "[data-testid=dateOfVaccineInput] [data-testid=formSelectMonth]";
+const dovDaySelector =
+    "[data-testid=dateOfVaccineInput] [data-testid=formSelectDay]";
```
Comment (MikeLyttle) [inline Testing/functional/tests/cypress/integration/ui/vaccineCard/publicCovidTest.js:27]: The public test page shouldn't have a date of vaccination.  I've created bug [AB#11901](https://dev.azure.com/qslvic/Health%20Gateway/_workitems/edit/11901) to track the issue.

File: Testing/functional/tests/cypress/integration/ui/vaccineCard/publicCovidTest.js [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+        select(dobMonthSelector, dobMonth);
+        select(dobDaySelector, dobDay);
+        select(dovYearSelector, dovYear);
+        select(dovMonthSelector, dovMonth);
+        select(dovDaySelector, dovDay);
+        clickCovidTestEnterButton();
+        cy.get("[data-testid=error-text-description]").should("be.visible");
+    });
+
+    it("Public Covid19 Test - Invalid Health Gateway Problem", () => {
```
Comment (MikeLyttle) [inline Testing/functional/tests/cypress/integration/ui/vaccineCard/publicCovidTest.js:360]: I would remove "Health Gateway Problem" from the name since it may confuse us later.

**Review by MikeLyttle — APPROVED** [review]

File: Apps/WebClient/src/ClientApp/src/views/publicCovidTest.vue [hunk 4]
```diff
@@ -515,40 +515,40 @@ export default class PublicCovidTestView extends Vue {
...
                                 <b-form-invalid-feedback
                                     v-if="
-                                        $v.dateOfVaccine.$dirty &&
-                                        !$v.dateOfVaccine.required
+                                        $v.dateOfCollection.$dirty &&
+                                        !$v.dateOfCollection.required
                                     "
-                                    aria-label="Invalid Date of Vaccine"
+                                    aria-label="Invalid Collection Date"
                                     data-testid="feedbackDovIsRequired"
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/views/publicCovidTest.vue:538]: DOV is short for date of vaccination and should be updated to refer to the collection date.

File: Testing/functional/tests/cypress/integration/ui/vaccineCard/publicCovidTest.js [hunk 5]
```diff
@@ -0,0 +1,406 @@
...
+    "[data-testid=dateOfCollectionInput] [data-testid=formSelectYear]";
+const dovMonthSelector =
+    "[data-testid=dateOfCollectionInput] [data-testid=formSelectMonth]";
+const dovDaySelector =
+    "[data-testid=dateOfCollectionInput] [data-testid=formSelectDay]";
+
+const feedbackPhnMustBeValidSelector = "[data-testid=feedbackPhnMustBeValid]";
+const feedbackPhnIsRequiredSelector = "[data-testid=feedbackPhnIsRequired]";
+const feedbackDobIsRequiredSelector = "[data-testid=feedbackDobIsRequired]";
+const feedbackDovIsRequiredSelector = "[data-testid=feedbackDovIsRequired]";
```
Comment (MikeLyttle) [inline Testing/functional/tests/cypress/integration/ui/vaccineCard/publicCovidTest.js:19]: See above comment regarding DOV.

File: Testing/functional/tests/cypress/integration/ui/vaccineCard/publicCovidTest.js [hunk 6]
```diff
@@ -0,0 +1,406 @@
...
+            .and("have.class", "form-control is-valid");
+    });
+
+    it("Public Covid19 Test - Invalid PHN via Select Year", () => {
+        enterCovidTestPHN(Cypress.env("phn").replace(/.$/, ""));
+        select(dobYearSelector, "Year");
+        cy.get(feedbackPhnMustBeValidSelector).should("be.visible");
+    });
+
+    it("Public Covid19 Test - PHN, DOB and DOV not entered via Click Enter", () => {
```
Comment (MikeLyttle) [inline Testing/functional/tests/cypress/integration/ui/vaccineCard/publicCovidTest.js:84]: See above comment regarding DOV for this and remaining tests.

**Review by MikeLyttle — CHANGES_REQUESTED** [review]

**Review by MikeLyttle — APPROVED** [review]

**Review by BrianMaki — APPROVED** [review]
